### PR TITLE
CLOUDP-163786: Add command "serverless backups snapshots list"

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Adds command "atlas serverless backups snapshots list"

_Jira ticket:_ CLOUDP-163786

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/store/cloud_provider_backup_serverless.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+// ServerlessSnapshots encapsulates the logic to manage different cloud providers.
+func (s *Store) ServerlessSnapshots(projectID, clusterName string, opts *atlas.ListOptions) (*atlas.CloudProviderSnapshots, error) {
+	o := &atlas.SnapshotReqPathParameters{
+		GroupID:      projectID,
+		InstanceName: clusterName,
+	}
+	switch s.service {
+	case config.CloudService, config.CloudGovService:
```
Comment (gssbzn) [inline internal/store/cloud_provider_backup_serverless.go:37]: I think gov doesn't support serverless

File: internal/cli/atlas/serverless/backup/snapshots/list.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		Use:     "list <clusterName>",
+		Short:   "Return all cloud backup snapshots for your project and serverless instance.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Read Only"),
+		Aliases: []string{"ls"},
+		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"clusterNameDesc": "Name of the Atlas serverless instance that contains the snapshots you want to retrieve.",
+		},
+		Example: fmt.Sprintf(`  # Return a JSON-formatted list of snapshots for the cluster named myDemo 
+  %s backups snapshots list myDemo --output json`, cli.ExampleAtlasEntryPoint()),
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/snapshots/list.go:73]:
```suggestion
  %s serverless snapshots list myDemo --output json`, cli.ExampleAtlasEntryPoint()),
```

File: internal/cli/atlas/serverless/backup/snapshots/list.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	cmd := &cobra.Command{
+		Use:     "list <clusterName>",
+		Short:   "Return all cloud backup snapshots for your project and serverless instance.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Read Only"),
+		Aliases: []string{"ls"},
+		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"clusterNameDesc": "Name of the Atlas serverless instance that contains the snapshots you want to retrieve.",
+		},
+		Example: fmt.Sprintf(`  # Return a JSON-formatted list of snapshots for the cluster named myDemo
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/snapshots/list.go:72]:
```suggestion
		Example: fmt.Sprintf(`  # Return a JSON-formatted list of snapshots for the instance named myDemo 
```
I think we prefer instance when talking about serverless

File: internal/cli/atlas/serverless/backup/snapshots/list_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 MongoDB Inc
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/snapshots/list_test.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: internal/cli/atlas/serverless/backup/snapshots/list.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 MongoDB Inc
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/snapshots/list.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: internal/cli/atlas/serverless/backup/backup_test.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2022 MongoDB Inc
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/backup_test.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: internal/cli/atlas/serverless/backup/backup.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 MongoDB Inc
```
Comment (gssbzn) [inline internal/cli/atlas/serverless/backup/backup.go:1]:
```suggestion
// Copyright 2023 MongoDB Inc
```

File: internal/cli/atlas/serverless/backup/snapshots/list.go [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+	return opts.Print(r)
+}
+
+// mongocli atlas serverless backups snapshots list <clusterName> [--projectId projectId] [--page N] [--limit N].
+func ListBuilder() *cobra.Command {
+	opts := new(ListOpts)
+	cmd := &cobra.Command{
+		Use:     "list <clusterName>",
+		Short:   "Return all cloud backup snapshots for your project and serverless instance.",
```
Comment (sarahsimpers) [inline internal/cli/atlas/serverless/backup/snapshots/list.go:65]:
```suggestion
		Short:   "Return all cloud backup snapshots for the specified serverless instance in your project.",
```

File: internal/cli/atlas/serverless/backup/snapshots/list.go [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Read Only"),
+		Aliases: []string{"ls"},
+		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"clusterNameDesc": "Name of the Atlas serverless instance that contains the snapshots you want to retrieve.",
```
Comment (sarahsimpers) [inline internal/cli/atlas/serverless/backup/snapshots/list.go:70]:
```suggestion
			"clusterNameDesc": "Label that identifies the Atlas serverless instance that contains the snapshots you want to return.",
```

**Review by sarahsimpers — CHANGES_REQUESTED** [review]
Small copy nitpicks. Thank you for including the required role! :)

**Review by sarahsimpers — APPROVED** [review]
LGTM, thanks!

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by wtrocki — APPROVED** [review]

**Pull request merged**